src/eventHandlers/checkSuite/index.spec.ts

```diff
@@ -334,8 +336,68 @@ describe('check Suite event handler', (): void => {
         },
       });
 
-    await checkSuiteHandle(octokit, 'dependabot-preview[bot]');
+    await checkSuiteHandle(octokit, 'dependabot-preview[bot]', 3);
 
     expect(warningSpy).toHaveBeenCalled();
   });
+
+  it('retries up to two times before failing', async (): Promise<void> => {
+    expect.assertions(6);
+
+    nock('https://api.github.com')
+      .post('/graphql')
+      .reply(OK, {
+        data: {
+          repository: {
+            pullRequest: {
+              commits: {
+                edges: [
+                  {
+                    node: {
+                      commit: {
+                        messageHeadline: COMMIT_HEADLINE,
+                      },
+                    },
+                  },
+                ],
+              },
+              id: PULL_REQUEST_ID,
+              mergeStateStatus: 'CLEAN',
+              mergeable: 'MERGEABLE',
+              merged: false,
+              reviews: {
+                edges: [
+                  {
+                    node: {
+                      state: 'APPROVED',
+                    },
+                  },
+                ],
+              },
+              state: 'OPEN',
+            },
+          },
+        },
+      });
+
+    const mergeSpy = jest
+      .spyOn(merge, 'merge')
+      .mockImplementation()
+      .mockRejectedValue(new Error('Error when merging'));
```

Comment:
We shouldn't be mocking internal implementations somewhere in the middle of the call chain of the test. Instead we should always strive to mock the outgoing call.

Reply:
Why not? Given the current use case, it doesn't matter which one, the outgoing call or a function, is mocked. The end result is the `merge` function failing with an error.

Reply:
The problem is that currently the mock is independent of the implementation. It is overriding the behavior and interface of the function. If the interface or implementation of the merge function changes, the tests wouldn't fail because it will consistently provide the same results because it is mocked.

What we want is that if the implementation or interface of the merge function changes, the test fail, so that we can correctly update the mock. So by mocking the response of the request (e.g. with nock), the merge function reacts to this change. Any change to the implementation of merge function will be picked up by the tests here in `checkSuite/index.spec.ts`.

Reply:
I don't fully agree. The test is focused on the retries and backoff strategy. These can only happen when the `merge` function fails, independent of its implementation or interface. From the perspective of this test, the `merge` function is a "black box": it simply needs to fail. 

Any change in the implementation or interface of the `merge` function should not affect this particular test because its implementation is not what is being tested. Of course, that being said, there are other tests that cover and validates the implementation of said function.

Reply:
The reason it is tested entirely top down for when that function passes is the same reason why it should be tested top down for when it fails. The contract of the merge function can change at any time in which case the test should be failing here. Because the merge function could be throwing, does not mean that it will in the future as the implementation might change. A hidden implementation assumes that it can throw, which is just an assumption. The only way to be sure is to test it top down.

Reply:
The `merge` function is being called only by `mergeWithRetry`, so a contract change wouldn't fail the test. An implementation change, such as not throwing an error, in the `merge` function requires a change in `mergeWithRetry`, invalidating the test as well. However, that's not explicit. Ok, I see what you mean. Thanks. Fixed in e28bc57.